agent/lib/broker_controller.js

```diff
@@ -73,6 +73,7 @@ BrokerController.prototype.on_connection_open = function (context) {
     var self = this;
     this.broker = new artemis.Artemis(context.connection);
     this.id = context.connection.container_id;
+    this.initializeGlobalMaxSizeFromBrokerIfUnset();
```

Comment:
Wouldn't it be better if this returned a promise and then you chained the actions that follow to the result?